src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1291,15 +1309,6 @@ def download(cls, pretrained_model_name, **kwargs) -> Union[str, os.PathLike]:
                         f"{candidate_file} as defined in `model_index.json` does not exist in {pretrained_model_name} and is not a module in 'diffusers/pipelines'."
                     )
 
-            if len(variant_filenames) == 0 and variant is not None:
```

Comment:
It's not an error, though. It's a deprecation. Do we exactly want to keep it that way? If so, we will have to remove it anyway because the deprecation is supposed to expire after "0.24.0" version. 

Instead, we are erroring out now from `from_pretrained()`:
https://github.com/huggingface/diffusers/blob/f583dad9bd4742b56264f236400bea1f71e74029/src/diffusers/pipelines/pipeline_utils.py#L757

Reply:
Ah got it. I think this should be resolved now. 

WDYT about catching these errors without having to download the actual files and leveraging `model_info()` (in case we're querying the Hub) or regular string matching (in case it's local)? Currently, we're still calling `download()` in case we don't have the model files cached. I think many errors can be caught and warnings can be thrown without having to do that. 

This could live in a future PR.